plugins/base/src/main/kotlin/signatures/KotlinSignatureProvider.kt

```diff
@@ -103,4 +107,21 @@ class KotlinSignatureProvider(ctcc: CommentsToContentConverter, logger: DokkaLog
             text("?")
         }
     }
+
+    fun PageContentBuilder.DocumentableContentBuilder.funType(type: KotlinTypeWrapper) {
+        val args = if (type.isExtension) {
+            type(type.arguments.first())
```

Comment:
You should not have if both modifying output AND returning a value! It's better to split it into two.

Reply:
Splitted.